libvast/src/system/configuration.cpp

```diff
@@ -79,32 +83,99 @@ caf::error configuration::parse(int argc, char** argv) {
   command_line.assign(argv + 1, argv + argc);
   // Move CAF options to the end of the command line, parse them, and then
   // remove them.
-  auto is_vast_opt = [](auto& x) {
-    return !(detail::starts_with(x, "--caf.")
-             || detail::starts_with(x, "--config=")
-             || detail::starts_with(x, "--config-file="));
-  };
+  auto is_vast_opt = [](auto& x) { return !detail::starts_with(x, "--caf."); };
   auto caf_opt = std::stable_partition(command_line.begin(), command_line.end(),
                                        is_vast_opt);
   std::vector<std::string> caf_args;
   std::move(caf_opt, command_line.end(), std::back_inserter(caf_args));
   command_line.erase(caf_opt, command_line.end());
-  for (auto& arg : caf_args) {
-    // Remove caf. prefix for CAF parser.
-    if (detail::starts_with(arg, "--caf."))
-      arg.erase(2, 4);
-    // Rewrite --config= option to CAF's expexted format.
+  // If the user provided a config file on the command line, we attempt to
+  // parse it last.
+  for (auto& arg : command_line)
     if (detail::starts_with(arg, "--config="))
-      arg.replace(8, 0, "-file");
+      config_paths.push_back(arg.substr(9));
+  // Parse and merge all configuration files.
+  caf::settings merged_settings;
+  for (const auto& config : config_paths) {
+    if (exists(config)) {
+      auto contents = load_contents(config);
+      if (!contents)
+        return contents.error();
+      auto yaml = from_yaml(*contents);
+      if (!yaml)
+        return yaml.error();
+      auto rec = caf::get_if<record>(&*yaml);
+      if (!rec)
+        return caf::make_error(ec::parse_error, "config file not a YAML map");
```

Comment:
Please improve this error context. From a user's point of view, this does not currently make any sense.

Reply:
Better?